interfaces/machine_parameters/ring.py

```diff
@@ -0,0 +1,588 @@
+                            'The synchronous data for all sections ' +
+                            'are defined time based and ' +
+                            'are not identical. This case is not yet ' +
+                            'implemented.')
+
+        # If the input synchronous data are by turn and not by time
+        # for all sections, these are simply combined.
+        if momentum_by_turn:
+            self.synchronous_data = ring_programs.momentum_program.combine_single_sections(
+                *momentum_list)
+
+        # Processing the momentum program
+        # Getting the options to get at which time samples the interpolation
+        # is made
+        t_start = kwargs.pop('t_start', 0)
+        t_stop = kwargs.pop('t_stop', np.inf)
+        interp_time = kwargs.pop('interp_time', 0)
+
+        # Treating the input of the interp_time option
+        if not hasattr(interp_time, '__iter__'):
+            interp_time = (interp_time, )
+
+        # Getting the sample times
+        sample_func, start, stop = tmng.time_from_sampling(*interp_time)
+
+        # Keeping only the values between the user defined bounds
+        if t_start > start:
+            start = t_start
+        if t_stop < stop:
+            stop = t_stop
+
+        # Getting options for the interpolation
+        interpolation = kwargs.pop('interpolation', 'linear')
+        store_turns = kwargs.pop('store_turns', True)
+
+        # Processing the momentum program and interpolating on the
+        # values defined by sample_func
+        momentum_processed = self.synchronous_data.preprocess(
+            self.Particle.mass,
+            self.circumference_design, sample_func,
+            interpolation, start, stop,
+            store_turns=store_turns)
+
+        # Getting the cycle time from the interpolation
+        self.cycle_time = momentum_processed[1]
+
+        # The machine turn numbers corresponding to the cycle_time
+        # are kept if the store_turns is enabled
+        if store_turns:
+            self.parameters_at_turn = self._parameters_at_turn
+            self.use_turns = momentum_processed[0].astype(int)
+        else:
+            self.parameters_at_turn = self._no_parameters_at_turn
+            self.use_turns = momentum_processed[0]
+
+        # Updating the number of turns in case it was changed after ramp
+        # interpolation
+        self.n_turns = momentum_processed.n_turns
+
+        # Storing the momentum program and computing all associated values
+        self.momentum = momentum_processed[2:]
+        self.beta = rt.mom_to_beta(
+            momentum_processed[2:], self.Particle.mass)
+        self.gamma = rt.mom_to_gamma(
+            momentum_processed[2:], self.Particle.mass)
+        self.energy = rt.momentum_to_energy(
+            momentum_processed[2:], self.Particle.mass)
+        self.kin_energy = rt.momentum_to_kin_energy(
+            momentum_processed[2:], self.Particle.mass)
```

Comment:
Needs `beta` and `gamma` subclasses of `_synchronous_data_program` adding, then these can all be done as:
```python
self.beta = self.momentum.to_beta(False, mass = self.Particle.mass)
# etc
```

Reply:
Corrected

For beta,gamma I cast those as np.arrays in the meanwhile to avoid confusion when printing